### ISSUE TITLE
Prevent server from returning too large responses

### ISSUE THREAD

**Issue by mokaddem** [MISP/MISP#6883]
- Adding a default pagination is *dangerous* as we will have lots of users not realizing they are missing data.
- Also depending on the returned format, silently adding the pagination may cause issues (especially for aggregation) 
- Proposed solution:
    - If the request is done without filters and the response becomes too large, an error should be returned (HTTP 403 - Resource requested too large)

**Comment by adulau** [comment]
Indeed. I would also propose in the error or return headers, a solution/example how to use pagination based on the query sent.

**Comment by chrisinmtown** [comment]
Please consider if you really want to use 403 -- Forbidden.  This isn't really a client error, the client cannot know the size of the response up front, so I don't think it's a client (4xx series) error. Instead I think it's a server (5xx series) error.  What about 507  - "insufficient storage"?  That seems closer to your intent here.

But what is "too large"? I should be able to download a gigabyte of data without much fuss, if that blows up my client that's my problem. And in a couple years, a gigabyte might be considered small.

Where will the definition of "large" be configured?

What will you do for methods like event/view (PyMISP method get_event) that currently offers absolutely no REST pagination options? What is a client-side user supposed to do when an event has 500,000 attributes on it?

**Comment by adulau** [comment]
I think @mokaddem means 413 not 403. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413

The issue is not too large but the processing time too which leads to time-out and so on.

**Comment by chrisinmtown** [comment]
Error code 413, as I understand it, means the client sent a REQUEST that is too large for the server's liking.  This error code does not seem like a good fit for the requirement you described, blocking a large RESPONSE.
